src/main/scala/com/gu/contentapi/firehose/kinesis/SingleEventProcessor.scala

```diff
@@ -0,0 +1,87 @@
+package com.gu.contentapi.firehose.kinesis
+
+import com.amazonaws.services.kinesis.model.Record
+import com.amazonaws.services.kinesis.clientlibrary.interfaces.{ IRecordProcessor, IRecordProcessorCheckpointer }
+import com.amazonaws.services.kinesis.clientlibrary.types.ShutdownReason
+import com.typesafe.scalalogging.LazyLogging
+import com.twitter.scrooge.ThriftStruct
+import java.util.concurrent.atomic.{ AtomicInteger, AtomicLong }
+import java.util.{ List => JList }
+import scala.collection.JavaConverters._
+import scala.concurrent.duration._
+import scala.util.{ Failure, Success }
+
+abstract class EventProcessor[EventT <: ThriftStruct](
+  checkpointInterval: Duration = 30.seconds,
```

Comment:
I'd remove the defaults from here, as you already have defaults for these values elsewhere. Multiple levels of default argument can get pretty confusing.